assembly/src/assembler/module_graph/mod.rs

```diff
@@ -50,8 +163,7 @@ pub struct ModuleGraph {
     ///
     /// Once added to the graph, modules become immutable, and any additional modules added after
     /// that must by definition only depend on modules in the graph, and not be depended upon.
-    #[allow(clippy::vec_box)]
-    pending: Vec<Box<Module>>,
+    pending: Vec<PendingModuleWrapper>,
```

Comment:
I don't know that we need to add non-AST modules to the pending set, as we don't need to do any post-processing of already-compiled modules. The main thing we need to do when adding a compiled library, is add them to `module_graph.modules` (so that they have a valid `ModuleIndex`), and then all of the procedures of that module can be added to the `CallGraph` as nodes (since once we have a `ModuleIndex`, we can derive a `GlobalProcedureIndex` for all of the procedures in that module).

This would remove the need for `PendingModuleWrapper` at least, AFAIK

Reply:
I think my initial goal here was not to break the module IDs of pending modules (from how `ModuleGraph::next_module_id()` works). For example, if we add a pending module, and then add a compiled module, then that will break the current way we allocate indices.

Were you thinking of refactoring that as well?